src/sizeMapping.js

```diff
@@ -1,61 +1,76 @@
+import { config } from 'src/config';
+
+let configuredLabels = [];
+let sizeConfig = [];
+
 /**
- * @module sizeMapping
+ * @typedef {object} SizeConfig
+ *
+ * @property {string} [mediaQuery] A CSS media query string that will to be interpreted by window.matchMedia.  If the
+ *  media query matches then the this config will be active and sizesSupported will filter bid and adUnit sizes.  If
+ *  this property is not present then this SizeConfig will only be active if triggered manually by a call to
+ *  pbjs.setConfig({labels:['label']) specifying one of the labels present on this SizeConfig.
+ * @property {Array<Array>} sizesSupported The sizes to be accepted if this SizeConfig is enabled.
+ * @property {Array<string>} labels The active labels to match this SizeConfig to an adUnits and/or bidders.
  */
-import * as utils from './utils';
-let _win;
 
-function mapSizes(adUnit) {
-  if (!isSizeMappingValid(adUnit.sizeMapping)) {
-    return adUnit.sizes;
-  }
-  const width = getScreenWidth();
-  if (!width) {
-    // size not detected - get largest value set for desktop
-    const mapping = adUnit.sizeMapping.reduce((prev, curr) => {
-      return prev.minWidth < curr.minWidth ? curr : prev;
-    });
-    if (mapping.sizes && mapping.sizes.length) {
-      return mapping.sizes;
-    }
-    return adUnit.sizes;
-  }
-  let sizes = '';
-  const mapping = adUnit.sizeMapping.find(sizeMapping => {
-    return width >= sizeMapping.minWidth;
-  });
-  if (mapping && mapping.sizes && mapping.sizes.length) {
-    sizes = mapping.sizes;
-    utils.logMessage(`AdUnit : ${adUnit.code} resized based on device width to : ${sizes}`);
-  } else {
-    utils.logMessage(`AdUnit : ${adUnit.code} not mapped to any sizes for device width. This request will be suppressed.`);
-  }
-  return sizes;
+/**
+ * @param {Array<string>} labels
+ */
+export function setLabels(labels) {
+  configuredLabels = labels;
 }
+config.getConfig('labels', config => setLabels(config.labels));
 
-function isSizeMappingValid(sizeMapping) {
-  if (utils.isArray(sizeMapping) && sizeMapping.length > 0) {
-    return true;
-  }
-  utils.logInfo('No size mapping defined');
-  return false;
+/**
+ *
+ * @param {Array<SizeConfig>} config
+ */
+export function setSizeConfig(config) {
+  sizeConfig = config;
 }
+config.getConfig('sizeConfig', config => setSizeConfig(config.sizeConfig));
```

Comment:
Trying to understand the usage of subscriber over here. 
I think that this can be also be done by 
`let sizeConfig = config.getConfig('sizeConfig') || []` 

Your thoughts on this ?

Reply:
subscribe method is easier to test IMO, because then I can just use given config subscribe function to set test config as opposed to stubbing the config.  Functionality wise they should behave similarly.